src/main/java/nl/knaw/dans/dvcli/command/CollectionAssignRole.java

```diff
@@ -45,42 +46,54 @@ public class CollectionAssignRole extends AbstractCmd {
     private CollectionCmd collectionCmd;
 
     static class SingleAssignment {
+        @Parameters(index = "1", paramLabel = "assignee", description = "The identifier of the user to assign the role to, example: @user")
+        String assignee;
+
         @Parameters(index = "0", paramLabel = "role", description = "The role to assign")
         String role;
 
-        @Parameters(index = "1", paramLabel = "assignee", description = "The identifier of the user to assign the role to")
-        String assignee;
     }
 
     static class AllArgs {
         @ArgGroup(exclusive = false)
         SingleAssignment singleAssignment;
 
-        @Option(names = { "-f", "--parameters-file" }, description = "CSV file to read parameters from. The file should have a header row with columns 'PID', 'ROLE', and 'ASSIGNEE'.")
+        @Option(names = { "-f", "--parameters-file" }, description = "CSV file to read parameters from. The file should have a header row with columns 'PID', 'ASSIGNEE', and 'ROLE'.")
         Path paramsFile;
     }
 
-    @ArgGroup(exclusive = true, multiplicity = "1")
+    @ArgGroup(multiplicity = "1")
     private AllArgs allArgs;
 
-    private record RoleAssignmentParams(DataverseApi collection, String role, String assignee) {
+    private record RoleAssignmentParams(DataverseApi collection, Optional<RoleAssignment> roleAssignment) {
     }
 
     private static class RoleAssignmentAction implements ThrowingFunction<RoleAssignmentParams, String, Exception> {
         @Override
         public String apply(RoleAssignmentParams roleAssignmentParams) throws IOException, DataverseException {
-            var assignment = new RoleAssignment();
-            assignment.setAssignee(roleAssignmentParams.assignee());
-            assignment.setRole(roleAssignmentParams.role());
-            var r = roleAssignmentParams.collection.assignRole(assignment);
-            return r.getEnvelopeAsString();
+            if (roleAssignmentParams.roleAssignment().isPresent()) {
+                var r = roleAssignmentParams.collection.assignRole(roleAssignmentParams.roleAssignment().get());
+                return r.getEnvelopeAsString();
+            }
+
+            return "Nothing to Do";
```

Comment:
Would be helpfull/informative to tell why there is nothing to do; like 'no roleassignment present'

Reply:
Changed.